### ISSUE TITLE
qwen3.5 (saracen9/amoral-qwen3.5-9b fine-tune): think=true reliably never produces content for non-trivial prompts, model emits done_reason=stop mid-thinking

### ISSUE THREAD

**Issue by darth-veitcher** [ollama/ollama#17777]
### What is the issue?

Using `saracen9/amoral-qwen3.5-9b` (a community fine-tune of the qwen3.5 architecture) via `/api/chat` with `"think": true`, the model reliably fails to ever produce `message.content` once the user/system instruction goes beyond a short, single-sentence request -- it generates a large amount of `message.thinking`, then stops (`done_reason: "stop"`) with `content` empty, well before hitting the model's own 8192-token `num_ctx` (context exhaustion is ruled out -- see reproduction below).

This may be related to (but is not the same regression as) #14745, which reports the same qwen3.5 architecture failing to properly complete a turn while in a special generation mode (tool-calling there; thinking-to-content transition here). That issue was reportedly fixed by downgrading to Ollama 0.17.5 from a 0.17.7 regression -- I'm running current Ollama (0.32.9), so this looks like either a resurfaced or a distinct regression in the same failure class.

### Reproduction (real, direct, live-tested)

A short, ~30-word single-message request with `think: true` and no system prompt reliably produces real, valid content (sometimes after 2-5 minutes of thinking, and with some quality variance).

The same model, given a longer (~130-270 word) instruction -- with or without a system-role message, doesn't matter, tested both ways -- fails 100% of the time across every real test I ran (4+ independent reproductions): thinking runs to 7,000-20,000 characters, then `done_reason: "stop"` fires with `content: ""`.

Minimal repro body (`POST /api/chat`):

\`\`\`json
{
  "model": "saracen9/amoral-qwen3.5-9b",
  "messages": [{"role": "user", "content": "Write the richest, most technically specific image-generation prompt you can for this seed intent. Stay faithful to it -- never invent unrelated subjects. Preserve any explicitly stated medium exactly. Elaborate richly on camera, lens, lighting, and composition choices to render the stated style. Never invent highly specific unstated detail like exact clothing or colors merely to look thorough.\n\nSeed intent: Elsa from Frozen, extreme facial closeup, hyper realistic real life photography. This is a real woman cosplaying as Elsa."}],
  "stream": false,
  "think": true
}
\`\`\`

Real response fields observed (non-streaming):
- \`done_reason\`: \`"stop"\`
- \`eval_count\`: \`1447\`
- \`prompt_eval_count\`: \`165\`
- \`message.thinking\`: 7217 characters
- \`message.content\`: \`""\`

1447 + 165 = 1612 tokens total -- nowhere near the model's own 8192-token \`num_ctx\` (confirmed via \`/api/show\`'s own \`parameters\` block: \`num_ctx 8192\`, architecture max \`context_length: 262144\`). This rules out context exhaustion as the cause -- the model is genuinely emitting its own stop token while still mid-thinking (or immediately after an incomplete/unclosed thinking block), never transitioning to content at all.

Also confirmed via a real streamed request (`"stream": true`) that this isn't a client-timeout artifact either: streaming a request with the same longer instruction shows continuous, healthy chunk arrival (~0.1s between chunks, zero idle gaps) for the full 200-400+ second duration, right up until `done: true` fires with the same empty-content result -- so this isn't a hung/stalled connection, the model is actively, continuously generating thinking tokens the whole time and simply never emits content.

### OS

macOS (also observed on a separate Linux RunPod-hosted instance running the same model)

### GPU

Apple Silicon (local); NVIDIA (RunPod instance)

### CPU

Apple Silicon (local)

### Ollama version

0.32.9

**Comment by rick-github** [comment]
The problem is that the model doesn't emit a `</think>` token, so all output is in the thinking field.  Reading the output shows a demarcation between thinking and the final prompt, but the demarcation is usually a markdown block (\`\`\`) or a line (`---`).  Since the same prompt works fine (emits a closing think token) with the base model (qwen3.5:9b), it would appear that the finetuning has broken the thinking phase.

**Comment by darth-veitcher** [comment]
Model owner here (`saracen9/amoral-qwen3.5-9b`). Thanks for the detailed report — I reproduced it and want to share findings that split this into two parts, one of which was on my end.

**Part 1 — my Modelfile made it much worse (now fixed).** My Modelfile shipped `PARAMETER presence_penalty 1.5` (inherited from an upstream abliterated Modelfile). A/B on ollama 0.32.9, `think: true`, a ~130-word prompt, no `num_predict` cap, empty-`content` rate over independent trials:

| `presence_penalty` | empty-content rate |
|---|---|
| 1.5 | ~75% (3/4) |
| 0.5 | still empties |
| 0.0 | ~17% (1/6) |
| 0.0 + `temperature 0.6` | still empties |

I've republished `:latest` with `presence_penalty 0.0` (Qwen3's thinking-mode default), which cuts it ~4×. **Anyone still hitting this on my model should re-pull.**

**Part 2 — a residual that looks like the parser.** Even at `presence_penalty 0` (and at `temperature 0.6`), it still fails intermittently: the model emits `<|im_end|>` **while still in the thinking channel**, and the turn ends with empty `content` / `done_reason: stop`, well under the context limit. It's strongly length-dependent (short prompts are fine), which fits a mechanism where the reasoning→content transition in the `qwen3.5` renderer/parser sometimes treats a mid-thinking end-of-turn token as the end of the turn. A high presence penalty just makes the model reach for that token sooner, which is why my Modelfile amplified it — but it isn't purely a sampling artifact. This may be the same class as #14745.

Happy to help test a parser fix if that's useful.

**Comment by rick-github** [comment]
Failure of the model to emit `</think>` is not a parser issue.
